tests/Unit/Requester/constructor.js

```diff
@@ -0,0 +1,287 @@
+        it('incorrect type of config.agentOptions.keepAlive, type = {type}', function (arg) {
+            assert.throws(
+                function () {
+                    new Requester({agentOptions: {keepAlive: arg.value}});
+                },
+                TypeError,
+                'config.agentOptions.keepAlive must be a boolean'
+            );
+        });
+    });
+
+    const invalidKeepAliveMsecsTypePayload = [null, true, '123', {}, ['a', 'b'], () => {}, Symbol()].map((type) => {
+        return {type: Object.prototype.toString.call(type), value: type};
+    });
+
+    dataDriven(invalidKeepAliveMsecsTypePayload, function () {
+        it('incorrect type of config.agentOptions.keepAliveMsecs, type = {type}', function (arg) {
+            assert.throws(
+                function () {
+                    new Requester({agentOptions: {keepAliveMsecs: arg.value}});
+                },
+                TypeError,
+                'config.agentOptions.keepAliveMsecs must be a positive integer'
+            );
+        });
+    });
+
+    const invalidKeepAliveMsecsValuePayload = [
+        {type: 'negative int', value: -1},
+        {type: 'not int but number', value: 1.2},
+        {type: 'Infinity', value: Infinity}
+    ];
+
+    dataDriven(invalidKeepAliveMsecsValuePayload, function () {
+        it('incorrect value of config.agentOptions.keepAliveMsecs, value = {type}', function (arg) {
+            assert.throws(
+                function () {
+                    new Requester({agentOptions: {keepAliveMsecs: arg.value}});
+                },
+                TypeError,
+                'config.agentOptions.keepAliveMsecs must be a positive integer'
+            );
+        });
+    });
+
+    const invalidMaxSocketsTypePayload = [null, true, '123', {}, ['a', 'b'], () => {}, Symbol()].map((type) => {
+        return {type: Object.prototype.toString.call(type), value: type};
+    });
+
+    dataDriven(invalidMaxSocketsTypePayload, function () {
+        it('incorrect type of config.agentOptions.maxSockets, type = {type}', function (arg) {
+            assert.throws(
+                function () {
+                    new Requester({agentOptions: {maxSockets: arg.value}});
+                },
+                TypeError,
+                'config.agentOptions.maxSockets must be a positive integer'
+            );
+        });
+    });
+
+    const invalidMaxSocketsValuePayload = [
+        {type: 'negative int', value: -1},
+        {type: 'not int but number', value: 1.2}
+    ];
+
+    dataDriven(invalidMaxSocketsValuePayload, function () {
+        it('incorrect value of config.agentOptions.maxSockets, value = {type}', function (arg) {
+            assert.throws(
+                function () {
+                    new Requester({agentOptions: {maxSockets: arg.value}});
+                },
+                TypeError,
+                'config.agentOptions.maxSockets must be a positive integer'
```

Comment:
constructor throws config.agentOptions.maxFreeSockets must be a positive integer or Infinity, so fix it pls and check why chai ignores mismatch.

Reply:
This test was deleted